test/gtest/uct/test_peer_failure.cc

```diff
@@ -314,10 +343,117 @@ UCS_TEST_P(test_uct_peer_failure_cb, desproy_ep_cb)
     check_caps(UCT_IFACE_FLAG_PUT_SHORT);
 
     wrap_errors();
-    kill_receiver0();
+    kill_receiver();
     EXPECT_EQ(uct_ep_put_short(ep0(), NULL, 0, 0, 0), UCS_OK);
     flush();
     restore_errors();
 }
 
 UCT_INSTANTIATE_TEST_CASE(test_uct_peer_failure_cb)
+
+class test_uct_peer_failure_multiple : public test_uct_peer_failure
+{
+public:
+    virtual void init();
+
+protected:
+    std::string txqlen_param_name() const;
+};
+
+void test_uct_peer_failure_multiple::init()
+{
+    const std::string   name = txqlen_param_name();
+    std::string         val;
+    size_t              tx_queue_len = 0;
+
+    if (get_config(name, val)) {
+        tx_queue_len = ucs::from_string<size_t>(val);
+        ASSERT_LT(0ul, tx_queue_len);
+    } else {
+        tx_queue_len = 256;
+        UCS_TEST_MESSAGE << name << " setting not found, "
+                         << "taken test default value: " << tx_queue_len;
+    }
+
+    if (GetParam()->tl_name == "rc_mlx5" || GetParam()->tl_name == "dc_mlx5") {
+        UCS_TEST_SKIP_R(GetParam()->tl_name + " temporary disabled");
+    }
+
+#if defined(__arm__) || defined(__aarch64__)
+    /* NOTE: Too much receivers causes failure of ibv_open_device */
+    test_uct_peer_failure::m_nreceivers = 10;
+#else
+    test_uct_peer_failure::m_nreceivers = tx_queue_len;
+#endif
+
+    test_uct_peer_failure::m_tx_window  = tx_queue_len / 3;
+
+    test_uct_peer_failure::init();
+
+    check_caps(UCT_IFACE_FLAG_AM_SHORT | UCT_IFACE_FLAG_PENDING);
+}
+
+std::string test_uct_peer_failure_multiple::txqlen_param_name() const
```

Comment:
i would encapsulate this as "size_t get_tx_queue_len()" - same function returns the actual value of the param